Name: Better Email Headers
Version: 3.0.5-2
Author: HaDAk
Description: Better Email Headers aims to compact the header fields in the email message view, so you can spend more time focusing on what's actually important.

--- .orig/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessageDisplay.js
+++ /media/cryptofs/apps/usr/palm/applications/com.palm.app.email/mail/source/MessageDisplay.js
@@ -43,7 +43,7 @@
 					]}
 				]},
 				{name: "senderInfoBlock", kind: "Control", className: "enyo-item first dotted-bottom", style: "position:relative;z-index:2;min-height: 46px; padding: 13px 14px 14px 11px", components: [
-					{kind: "VFlexBox", flex: 1, pack:"center", className: "mail-body-from-box", onclick: "showFromContactDetails", components: [									
+					{kind: "VFlexBox", flex: 1, pack: "center", className: "mail-body-from-box", components: [								
 						{kind:"HFlexBox", components: [
                             {name:"avatarBox", showing:true, width:"121px", height:"121px", className:"avatar-image-box", components:[
 								{kind: "Image", name:"avatarImage", width:"105px", height:"105px", style:"padding:8px;", src: this.DEFAULT_SENDER_IMAGE}
@@ -45,8 +45,8 @@
 				{name: "senderInfoBlock", kind: "Control", className: "enyo-item first dotted-bottom", style: "position:relative;z-index:2;min-height: 46px; padding: 13px 14px 14px 11px", components: [
 					{kind: "VFlexBox", flex: 1, pack:"center", className: "mail-body-from-box", onclick: "showFromContactDetails", components: [									
 						{kind:"HFlexBox", components: [
-                            {name:"avatarBox", showing:true, width:"121px", height:"121px", className:"avatar-image-box", components:[
-								{kind: "Image", name:"avatarImage", width:"105px", height:"105px", style:"padding:8px;", src: this.DEFAULT_SENDER_IMAGE}
+                            {name: "avatarBox", showing: true, width: "76px", height: "76px", className: "avatar-image-box", onclick: "showFromContactDetails", components: [
+                                {kind: "Image", name: "avatarImage", width: "68px", height: "68px", style: "padding:4px;", src: this.DEFAULT_SENDER_IMAGE}
 							]},
 							{kind:"VFlexBox", flex: 1, pack:"bottom", components: [
 								{kind:"HFlexBox", flex: 1, pack:"bottom", components: [
@@ -50,8 +50,17 @@
 							]},
 							{kind:"VFlexBox", flex: 1, pack:"bottom", components: [
 								{kind:"HFlexBox", flex: 1, pack:"bottom", components: [
-									{flex:1, showing:true, components: [/* Alternate location for onlineStatus */ ]},
-									{kind:"Control", className:"time-box", components: [
+                                    {flex: 1, showing: true, components: [
+                                        {name: "from", className: "mail-body-from", content: "&nbsp;", onclick: "showFromContactDetails", flex: 1}, // this is the sender name
+                                        {name: "from_addr", className: "mail-body-from-addr", content: "&nbsp;", onclick: "showFromContactDetails", flex: 1}, // this is the email address
+                                        {kind: "VFlexBox", flex: 1, style: "height: 25px;", components: [
+                                                {name: "onlineStatus", showing: false, style: "position: absolute; top: 10px; left: 75px; z-index: 20;", kind: "Image", src: ""},
+                                                
+                                                {name: "to", kind: "", allowHtml: true, className: "mail-body-recipients", onclick: "toggleRecipients"}, // This is the to: line
+
+                                            ]}
+                                    ]},									
+                                    {kind:"Control", className:"time-box", components: [
 										{name: "whenMonth", className: "mail-when-month"},
 										{name: "whenDay", className: "mail-when-day"},
 										{name: "whenTime", className: "mail-when-time"}
@@ -56,12 +65,6 @@
 										{name: "whenDay", className: "mail-when-day"},
 										{name: "whenTime", className: "mail-when-time"}
 									]}
-								]},
-								{kind:"HFlexBox", flex: 1, style:"padding-top:16px", components: [
-									{kind:"VFlexBox", pack:"center", flex:0, components: [
-										{name: "onlineStatus", showing: false, style:"padding:18px 0px 0px 0px;", kind: "Image", src: ""}
-									]},
-									{name: "from", className: "mail-body-from", content: "&nbsp;", flex: 1}
 								]}
 							]}
 						]}	
@@ -67,12 +70,7 @@
 						]}	
 					]}							
 				]},
-				{name: "toItem", layoutKind: "HFlexLayout", style: "position:relative;z-index:2;min-height: 50px; padding: 4px 14px 4px 14px", kind: "Item", className:"dotted-bottom", showing: false, onclick: "toggleRecipients", components: [
-					{name: "to", kind:"", flex: 1, allowHtml:true, className: "mail-body-recipients"},
-					{name:"ccBccButton", showing: true, kind:"Button", className:"enyo-button-light", width:"14px", height:"20px", style: "", components: [
-						{kind:"Image", style: "margin:  -6px 0px 0px -11px;", src:"../images/drawer_arrow.png"}
-					]}
-				]},
+
 				// if changing max-height, update value in this.resize as well
 				{name: "recipients", style:"background-color:#EEEEEE; font-size: 16px", kind: "Drawer", open: false, onOpenChanged: "resize", onOpenAnimationComplete: "resize", components: [
 					{name: "recipientsScroll", kind:"BasicFadeScroller", autoVertical: true, vertical: false, style: "max-height:300px;", onclick: "holdyClick", components: [
@@ -82,10 +80,9 @@
 						]}
 					]}	
 				]},
-				{kind:"HFlexBox", name:"subjectBox", align: "center", className: "mail-body-subject", components: [
-                    {name:"subjectLabel", className: "mail-body-subject-label", content: $L("Subject:")},
-					{name: "subject", flex: 1, width: 0, allowHtml: true, onPrevious: "doPrevious", style:'line-height:22px', onNext: "doNext"}
-				]},
+
+                // This is the subject line
+                {name: "subject", flex: 1, width: 0, allowHtml: true, onPrevious: "doPrevious", className: "mail-body-subject dotted-bottom", style: "position: relative;z-index:2;", onNext: "doNext"},
 				{name: "attachments", style: "", kind: "AttachmentsDrawer", collapsible:true, onInlineDownloaded: "_loadMessage", onOpenAnimationComplete: "resize"},
 				
 				/* TODO meetingInfo needs to be styled appropriately:
@@ -600,7 +597,7 @@
 			}
 			return 0;
 		});
-		this.$.toItem.setShowing(this.recipients.length);
+		//this.$.toItem.setShowing(this.recipients.length);
 		this.$.to.setContent(this.makeRecipientsSummary());
 		this.$.recipientsList.renderContent();
 
@@ -756,6 +753,7 @@
 	_hookupSenderName: function(contactDetails) {
 		var emailData = this.email.data;
 		var base = emailData.from.name || emailData.from.addr;
+		var base1 = emailData.from.addr;
 		this.$.from.setContent(contactDetails.getName() || base);
 	},
 	
@@ -757,6 +755,7 @@
 		var emailData = this.email.data;
 		var base = emailData.from.name || emailData.from.addr;
 		this.$.from.setContent(contactDetails.getName() || base);
+		this.$.from_addr.setContent(base1);
 	},
 	
 	showMessageError: function(divName) {
--- .orig/media/cryptofs/apps/usr/palm/applications/com.palm.app.email/css/mail.css
+++ /media/cryptofs/apps/usr/palm/applications/com.palm.app.email/css/mail.css
@@ -306,16 +306,29 @@
 }
 
 .mail-body-from {
-	font-size:34px;
-	font-family: PreludeWGL-Light;
-	overflow: hidden;
-	margin-right: 10px;
-	margin-left: 6px;
-	overflow: hidden;
-	text-overflow: ellipsis;
-	white-space:nowrap;
-	color: #333333;
-	padding-top:10px;
+    position: relative;
+    top: -10px;
+    font-size: 36px;
+    font-family: PreludeWGL-Light;
+    overflow: hidden;
+    margin-right: 10px;
+    margin-left: 2px;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    color: #333333;
+}
+
+.mail-body-from-addr {
+    position: relative;
+    top: -15px;
+    font-size: 16px;
+    margin-left: 2px;
+    margin-right: 10px;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    color: gray;
 }
 
 .mail-body-recipients-caption {
@@ -321,11 +334,7 @@
 .mail-body-recipients-caption {
 	color: #2d6c94;
 	display:inline;
-/*	border-width: 0px 4px;
-	line-height: 38px;
-	height: 38px;
-	min-width: 38px;
-	text-align: center;*/
+	text-align: center;
 	margin: 0 8px 0 0;
 	/*-webkit-border-image: url(../images/square-button-sprite.png) 0 4 40 4;*/
 }
@@ -336,11 +345,14 @@
 }
 
 .mail-body-recipients {
-	font-size: 16px;
-	line-height: 40px;
-	overflow: hidden;
-	white-space: nowrap;
-	text-overflow: ellipsis;
+    position: relative;
+    top: -15px;
+    font-size: 16px;
+    margin-left: 2px;
+    margin-right: 10px;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
 }
 
 .mail-body-recipients.expanded {
@@ -348,11 +360,14 @@
 }
 
 .mail-body-subject {
-	color: black;
-	font-size: 16px;
-	/*background: transparent url(../images/command-menu-bg.png) left -10px repeat-x;*/
-	padding: 3px 0;
-	border-bottom: 1px solid #ccc;
+    color: #2d6c94;
+    font-size: 24px;
+    /*background: transparent url(../images/command-menu-bg.png) left -10px repeat-x;*/
+    padding: 3px 0;
+    border-bottom: 1px solid #ccc;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
 }
 
 .mail-body-subject-label {
@@ -652,7 +667,7 @@
 .mail-when-day {
 	font-family: PreludeWGL-Light;
 	color:#333;
-	font-size:38px;
+	font-size:36px;
 	border-bottom:1px solid #DEDEDE;
 	font-weight:lighter;
 	margin:-3px -4px 0 -4px;
@@ -656,7 +671,7 @@
 	border-bottom:1px solid #DEDEDE;
 	font-weight:lighter;
 	margin:-3px -4px 0 -4px;
-	line-height:45px;
+	line-height:42px;
 }
 
 .mail-when-time {
\ No newline at end of file
